### PR TITLE
Slugs for services 447

### PR DESCRIPTION
**Pull request by MikeyManoguerra**
This adds a slug field for the service.  
 the django admin allows the user to create a slug field, but then it is un-editable.
its also un-editable in the serializer, which means its set for life once created.

The result is we can use this as a unique identifier for services. we can also use this to create dynamic route requests in the future, which will help for an editable form

### CODE REVIEW

**Comment by MikeyManoguerra** [comment]
i am making this as a draft bc i want to rebase once jack gets his most recent PR merged in. then i will wipe the migrations, since i edited them directly to get this working in development.

**Comment by JackRyan1989** [comment]
@MikeyManoguerra I have merged in my latest PR. I'll wait for you to wipe the migrations and then I'll take a look at this review.

**Comment by MikeyManoguerra** [comment]
@JackRyan1989 ok its ready

File: core/models/program.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 from django.db import models
-
+import re
```
Comment (JackRyan1989) [inline core/models/program.py:2]: Why import re?

File: core/programs/views.py [hunk 2]
```diff
@@ -1,6 +1,6 @@
 from core.viewsets import ModelViewSet
 from rest_framework.response import Response
-from core.models import Program, Service
+from core.models import Program, Program
```
Comment (JackRyan1989) [inline core/programs/views.py:3]: Why import Program twice?

**Review by JackRyan1989 — APPROVED** [review]
Two questions, neither are critical nor should they hold up this PR.

File: core/management/commands/seed.py [hunk 3]
```diff
@@ -253,13 +254,15 @@ def create_programs(output=True):
...
+            s.slug = f"{p.name}-{s.name}"
+
             s.available = random_bool()
             s.program = p
             s.full_clean()
             s.save()
 
             if output:
-                print("Created {}: '{}'".format(p.name, s.name))
+                print("Created {}: '{}'".format(p.slug, s.slug))
```
Comment (alex-cots) [inline core/management/commands/seed.py:265]: I might be missing something but I don't think `p` has a slug attribute

**Review by alex-cots — CHANGES_REQUESTED** [review]
Backend looks good. I think just the one line in seed is a small mistake, plus Jack's comments

File: core/management/commands/seed.py (hunk 3 above)
Comment (MikeyManoguerra) [reply core/management/commands/seed.py:265]: ah dang you're right, i don't know how that slipped back in.

File: core/programs/views.py (hunk 2 above)
Comment (MikeyManoguerra) [reply core/programs/views.py:3]: weird no idea

File: core/models/program.py (hunk 1 above)
Comment (MikeyManoguerra) [reply core/models/program.py:2]: i was going to add a progam slug then decided against it. will remove

**Review by alex-cots — APPROVED** [review]

**Review by JackRyan1989 — APPROVED** [review]

**Pull request merged**